### PR TITLE
fix auto login, AUTO_LOGIN -> AUTO_LOGIN_REGEX

### PR DESCRIPTION
**Pull request by patte**
AUTO_LOGIN_REGEX signs in matching emails automatically after 2 seconds.
Set `AUTO_LOGIN_REGEX=^([a-zA-Z0-9._%+-]+)@republik\.ch$` for a more open life in development.

- fix auto `authorizeSession`
- rename AUTO_LOGIN -> AUTO_LOGIN_REGEX

screenshot from 2019-02-12 01-25-44

### CODE REVIEW

File: packages/auth/lib/Users.js [hunk 1]
```diff
@@ -205,15 +207,14 @@ const signIn = async (_email, context, pgdb, req, consents, _tokenType) => {
...
-    // - email addresses containing «not» will neither be logged in nor send an sign request
-    const testMatch = email.match(/^([a-zA-Z0-9._%+-]+)@test\.project-r\.construction$/)
-    if (testMatch) {
-      if (testMatch[1].indexOf('not') === -1) {
-        return true
-      }
+  if (AUTO_LOGIN_REGEX) {
+    let testMatch = false
+    try {
+      testMatch = new RegExp(AUTO_LOGIN_REGEX).test(email)
```
Comment (patrickvenetz) [inline packages/auth/lib/Users.js]: Maybe `return` early. Don't see (yet) why it should be stuffed into variable.

File: packages/auth/lib/Users.js [hunk 2]
```diff
@@ -205,15 +207,14 @@ const signIn = async (_email, context, pgdb, req, consents, _tokenType) => {
...
+    } catch (e) {
+      testMatch = false
```
Comment (patrickvenetz) [inline packages/auth/lib/Users.js]: Maybe `return false` and get rid of `testMatch` variable all together.

File: packages/auth/lib/Users.js [hunk 3]
```diff
@@ -173,11 +173,13 @@ const signIn = async (_email, context, pgdb, req, consents, _tokenType) => {
     })
     if (shouldAutoLogin({ email })) {
       setTimeout(async () => {
-        console.log('AUTO_LOGIN!')
+        console.warn(`🔓💥 auto login for ${email}!!! AUTO_LOGIN_REGEX`)
```
Comment (patrickvenetz) [inline packages/auth/lib/Users.js]:
```suggestion
        console.warn(`🔓💥 Auto Login for ${email} due to AUTO_LOGIN_REGEX`)
```

File: packages/auth/lib/Users.js [hunk 4]
```diff
@@ -173,11 +173,13 @@ const signIn = async (_email, context, pgdb, req, consents, _tokenType) => {
...
         await authorizeSession({
           pgdb,
           tokens: [token],
-          email
+          email,
+          req,
```
Comment (patrickvenetz) [inline packages/auth/lib/Users.js]: Was this broken before adding `req` and `me`?

File: .travis.yml [hunk 5]
```diff
@@ -44,6 +44,7 @@ env:
     - PARKING_PLEDGE_ID=00000000-1234-0000-0000-000000000001
     - PARKING_USER_ID=00000000-4321-0000-0000-000000000001
     - DISPLAY_AUTHOR_SECRET=testtest
+    - AUTO_LOGIN_REGEX=^([a-zA-Z0-9._%+-]+)@test\.project-r\.construction$
```
Comment (patrickvenetz) [inline .travis.yml]: Remove `AUTO_LOGIN` further above. It appears that `AUTO_LOGIN` is no more.

File: packages/auth/lib/Users.js (hunk 4 above)
Comment (patte) [reply packages/auth/lib/Users.js]: yes, and therefore the tests as well

**Pull request merged**